src/category_theory/quotient.lean

```diff
@@ -0,0 +1,101 @@
+import category_theory.functor
+import category_theory.eq_to_hom
+
+/-
+  This file defines quotient categories. Given a category and an arbitrary family of relations
+  on its homsets, thought of as identifying some homs, we construct a new category on an
+  identical type of objects, where related homs are identified.
+
+    Note that we do *not* assume that our relation behaves well under composition, or that it is
+  an equivalence relation.
+-/
```

Comment:
Start the "module doc" with a `/-!` so that the doc generation tools will make it look nice. A markdown title is also good here. Have a look at some other files.

Perhaps explicitly mention in the module doc that you introduce a type synonym for the objects in the quotient category, rather than just saying "identical type".